treeherder/perf/models.py

```diff
@@ -143,34 +131,35 @@ def cycle_data(self, repository, cycle_interval, chunk_size, sleep_time,
         # also remove any signatures which are (no longer) associated with
         # a job
         logger.warning('Removing performance signatures with missing jobs...')
-        for signature in PerformanceSignature.objects.filter(
-                repository=repository):
+        for signature in PerformanceSignature.objects.all():
+            if self._should_quit(started_at, max_overall_runtime):
```

Comment:
I'm curious why time it and end it this way versus catching a timeout exception?

Reply:
> I'm curious why time it and end it this way versus catching a timeout exception?

Indeed, this would make the code more readable.